geode-core/src/main/java/org/apache/geode/cache/client/internal/ClientSideHandshakeImpl.java

```diff
@@ -355,7 +355,10 @@ private byte write(DataOutputStream dos, DataInputStream dis, CommunicationMode
         // for testing
         VersioningIO.writeOrdinal(hdos, overrideClientVersion, true);
       } else {
-        VersioningIO.writeOrdinal(hdos, currentClientVersion.ordinal(), true);
+        VersioningIO.writeOrdinal(hdos,
+            communicationMode.isWAN() ? KnownVersion.CURRENT_ORDINAL
```

Comment:
More for curiosity but why can't WAN utilize the version checking?

Reply:
that I don't know.  We may want to change that, but WAN does some fancy & mysterious things in the handshake code and elsewhere and I wasn't sure it would be okay to roll its comms back to 1.8.

Reply:
Yeah, when I was going through pulling out a bunch of the really old version checks there were lots of checks for WAN. I suspected it was for some form of cross version compatibility between sites but it wasn't clear. Perhaps @boglesby could shed some light.